src/openms/source/QC/FragmentMassError.cpp

```diff
@@ -274,19 +286,80 @@ namespace OpenMS
     // computes variance
     fmap.applyFunctionOnPeptideIDs(fVar);
 
-    result.variance_ppm = result.variance_ppm / counter_ppm;
-
     results_.push_back(result);
 
   }
 
+  void FragmentMassError::compute(std::vector<PeptideIdentification>& pep_ids, const ProteinIdentification::SearchParameters& search_params, const MSExperiment& exp, const QCBase::SpectraMap& map_to_spectrum, ToleranceUnit tolerance_unit, double tolerance)
+  {
+    Statistics result;
+
+    if (pep_ids.empty())
+    {
+      results_.push_back(result);
+      return;
+    }
+    // accumulates ppm errors over all first PeptideHits
+    double accumulator_ppm{};
+
+    // counts number of ppm errors
+    UInt32 counter_ppm{};
+
+    //---------------------------------------------------------------------
+    // Prepare MSExperiment
+    //---------------------------------------------------------------------
+
+    // filter settings
+    WindowMower window_mower_filter;
+    Param filter_param = window_mower_filter.getParameters();
+    filter_param.setValue("windowsize", 100.0, "The size of the sliding window along the m/z axis.");
+    filter_param.setValue("peakcount", 6, "The number of peaks that should be kept.");
+    filter_param.setValue("movetype", "jump", "Whether sliding window (one peak steps) or jumping window (window size steps) should be used.");
+    window_mower_filter.setParameters(filter_param);
+
+    //-------------------------------------------------------------------
+    // find tolerance unit and value
```

Comment:
please document that the tolerance parameter is overwritten if ToleranceUnit::AUTO is chosen. could be a bit surprisong